src/services/filament/spool-manager.octoprint.service.ts

```diff
@@ -28,11 +28,15 @@ export class SpoolManagerOctoprintService implements FilamentPluginService {
   public getCurrentSpools(): Observable<Array<FilamentSpool>> {
     return this.callSpoolManagerAPI('hideInactiveSpools', 0, 3000, 'lastUse', 'desc').pipe(
       map((spools: SpoolManagerSpoolList): FilamentSpool[] => {
-        if (spools.selectedSpools[0]) {
-          return spools.selectedSpools.map(s => this.convertFilamentManagerSpool(s));
-        } else {
-          return null;
-        }
+        const sourceSpools = spools.selectedSpools?.some(spool => spool !== null)
+          ? spools.selectedSpools
+          : spools.allSpools?.slice(0, 5) || [];
+
+        return sourceSpools.length > 0
+          ? sourceSpools
+              .map((spool, index) => (spool ? this.convertFilamentManagerSpool(spool, index + 1) : null))
```

Comment:
Can the `id + 1` be done in the MMU service? Tweaks for MMU compatibility should be limited to the scope of MMU code.